ydb/docs/en/core/integrations/orm/linq2db.md

```diff
@@ -0,0 +1,503 @@
+
+{% list tabs group=lang %}
+
+- C#
+
+  ```csharp
+  using System;
+  using LinqToDB.Mapping;
+
+  [Table("employee")]
+  [Index("full_name", Name = "employee_full_name_idx")]
+  public class Employee
+  {
+      [PrimaryKey] public long Id { get; set; }
+
+      [Column("full_name"), NotNull] public string FullName { get; set; } = null!;
+
+      [Column, NotNull] public string Email { get; set; } = null!;
+
+      [Column("hire_date", DataType = DataType.Date), NotNull]
+      public DateTime HireDate { get; set; }
+
+      [Column(DataType = DataType.Decimal, Precision = 22, Scale = 9), NotNull]
+      public decimal Salary { get; set; }
+
+      [Column("is_active"), NotNull] public bool IsActive { get; set; }
+
+      [Column, NotNull] public string Department { get; set; } = null!;
+
+      [Column, NotNull] public int Age { get; set; }
+  }
+  ```
+
+{% endlist %}
+
+**Generated DDL**:
+
+```yql
+CREATE TABLE employee (
+    Id Int64 NOT NULL,
+    full_name Utf8 NOT NULL,
+    Email Utf8 NOT NULL,
+    hire_date Date NOT NULL,
+    Salary Decimal(22,9) NOT NULL,
+    is_active Bool NOT NULL,
+    Department Utf8 NOT NULL,
+    Age Int32 NOT NULL,
+    PRIMARY KEY (Id)
+);
+
+ALTER TABLE employee
+  ADD INDEX employee_full_name_idx GLOBAL
+       ON (full_name);
+```
+
+Usage example
+
+```csharp
+using System;
+using System.Linq;
+using LinqToDB;
+using LinqToDB.Data;
+
+using var db = new DataConnection("YDB");
+
+// INSERT
+var employee = new Employee
+{
+    Id         = 1L,
+    FullName   = "Example",
+    Email      = "example@bk.com",
+    HireDate   = new DateTime(2023, 12, 20),
+    Salary     = 500000.000000000m,
+    IsActive   = true,
+    Department = "YDB AppTeam",
+    Age        = 23,
+};
+db.Insert(employee);
+
+// SELECT by primary key
+var loaded = db.GetTable<Employee>()
+               .FirstOrDefault(e => e.Id == employee.Id);
+
+// UPDATE Email/Department/Salary by primary key
+db.GetTable<Employee>()
+  .Where(e => e.Id == employee.Id)
+  .Set(e => e.Email,      "example+updated@bk.com")
+  .Set(e => e.Department, "Analytics")
+  .Set(e => e.Salary,     550000.000000000m)
+  .Update();
+
+// DELETE by primary key
+db.GetTable<Employee>()
+  .Where(e => e.Id == employee.Id)
+  .Delete();
+```
+
+**YQL samples generated by the provider for simple operations:**
+
+- Insert a single row
+
+  ```yql
+  INSERT INTO employee (Age,Department,Email,full_name,hire_date,is_active,Salary,Id)
+  VALUES (?,?,?,?,?,?,?,?);
+  ```
+
+- Read by primary key
+
+  ```yql
+  SELECT
+      e.Id, e.full_name, e.Email, e.hire_date, e.Salary, e.is_active, e.Department, e.Age
+  FROM employee AS e
+  WHERE e.Id = ?;
+  ```
+
+- Update by primary key
+
+  ```yql
+  UPDATE employee
+  SET
+      Email      = ?,
+      Department = ?,
+      Salary     = ?
+  WHERE Id = ?;
+  ```
+
+> Note: The vendor specifically removes the alias from the update table to comply with the YQL rules. This is done by the provider's SQL optimizer.
```

Comment:
It sounds like an implementation detail. I'd be more interested in why it generates `?` instead of `DECLARE` and how to do less standard queries like `UPSERT`.